### PR TITLE
src: maybe experimental implementation of the unified approved licenses list check

### PR DESCRIPTION
**Pull request by helio-frota**
Connects to #54

### CODE REVIEW

**Comment by helio-frota** [comment]
usage : `licenser --unifiedlist` 
for now this will show a warning for the `approved` only.
Maybe is good to get some feedback before continue on black list and or if we need to worry with 
some option to specify another `unified json file`, since we already are using the file with approved yes/no property...

File: lib/unifiedlist.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        lf.license === a['fedora_name'] ||
+        lf.license === a['spdx_abbrev'] ||
+        lf.license === a['spdx_name']) {
+        approvedFound.add(lf);
+      }
+    });
+  });
+
+  if (approvedFound.size > 0) {
+    console.error(`========= WARNING APPROVED LICENSES ==========`);
```
Comment (danbev) [inline lib/unifiedlist.js]: Should this really be a warning message?

Comment (helio-frota) [inline lib/unifiedlist.js]: fixed but please see the comments on PR.

**Comment by helio-frota** [comment]
@danbev 

Also this PR includes `not approved licenses` and when the user choose the `unified list` option the program will not show the `white and black lists` . see:

Before:
```
========= WARNING WHITE-LISTED LICENSES ==========
name: node-builtins , version: 0.1.1 , licenses: Apache-2.0
name: roi , version: 0.15.0 , licenses: Apache*
========= WARNING WHITE-LISTED LICENSES ==========
Dependencies found in package.json but not in xml: commander,decomment
Please run 'licenser --ignore-version-range' to show all declared dependencies on generated xml.
========= APPROVED LICENSES ==========
name: node-builtins , version: 0.1.1 , licenses: Apache-2.0
========= APPROVED LICENSES ==========
========= NOT APPROVED LICENSES ==========
name: roi , version: 0.15.0 , licenses: Apache*
========= NOT APPROVED LICENSES ==========
```

After:
```
Dependencies found in package.json but not in xml: commander,decomment
Please run 'licenser --ignore-version-range' to show all declared dependencies on generated xml.
========= APPROVED LICENSES ==========
name: node-builtins , version: 0.1.1 , licenses: Apache-2.0
========= APPROVED LICENSES ==========
========= NOT APPROVED LICENSES ==========
name: roi , version: 0.15.0 , licenses: Apache*
========= NOT APPROVED LICENSES ==========
```

**Note**

I've changed manually the roi dependency from szero from `Apache-2.0` to `FooApache-2.0`  and the licenser added ** * ** to the value

**Comment by danbev** [comment]
> Also this PR includes not approved licenses and when the user choose the unified list option the program will not show the white and black lists

Nice, I think that makes sense. 👍 



> I've changed manually the roi dependency from szero from Apache-2.0 to FooApache-2.0 and the licenser added * to the value
> 
> name: roi , version: 0.15.0 , licenses: Apache*
> 
> Question: Is it considered Not approved ?

I'm not sure what the `*` means here. I was expecting it to report `FooApache-2.0`. I might be missing some context regarding the requirements on this as I've been away.

**Comment by helio-frota** [comment]
> I'm not sure what the * means here. I was expecting it to report FooApache-2.0. I might be missing some context regarding the requirements on this as I've been away.

Yah the licenser-checker in some moment detects Apache-2.0, by reading LICENSE or README or COPYING and sets the value to Apache*

**Comment by danbev** [comment]
> Yah the licenser-checker in some moment detects Apache-2.0, by reading LICENSE or README or COPYING and sets the value to Apache*

Ah I see. I don't think this is a major concern as we are still picking up on that the license is not approved.

**Pull request merged**